.github/workflows/build.yml

```diff
@@ -104,8 +113,9 @@ jobs:
           echo ::set-output name=shield-arg::${SHIELD_ARG}
           echo ::set-output name=artifact-name::${ARTIFACT_NAME}
       - name: Build (west build)
-        run: west build -s app -b ${{ matrix.board }} -- ${{ steps.variables.outputs.shield-arg }}
+        run: west build -s app -b ${{ matrix.board }} -- ${{ steps.variables.outputs.shield-arg }} ${{ matrix.build-arg }}
       - name: Archive artifacts
+        if: ${{ !matrix.skip-archive }}
```

Comment:
What's the benefit of adding archive skipping?  Is this strictly necessary?

Reply:
Without this, the archive and uf2 images will be the same, and so we'd replace the previously archived version with default settings w/ this overridden one w/ various optional things turned on.

Reply:
Oh right, I didn't realise it overwrote artifacts with the same name:
https://github.com/actions/upload-artifact#uploading-to-the-same-artifact

Yeah, then I guess this is the simplest approach.

Reply:
Alternatively we could provide a unique suffix for the artifact name.  That might also prove more flexible going forward.

Reply:
We can look at doing that, I don't see the value (yet) in archiving these though, these are for test coverage, not "use this image"